src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -159,6 +185,66 @@ public void setup() {
     }
   }
 
+  /**
+   * Utility method that attempts to determine the token source based on user provided parameters.
+   *
+   * @return {@link SplunkTokenSource}
+   */
+  private SplunkTokenSource resolveTokenSource() {
+
+    if (tokenSecretId() != null && tokenSecretId().get() != null) {
```

Comment:
I would exclude the `SECRET_MANAGER` branch from `resolveTokenSource`. The idea is that this method should support only the already existing token sources (i.e. PLAINTEXT and KMS), for reverse compatibility.
If someone wants to use the new `SECRET_MANAGER` source, they will **have** to set `tokenSource` as well (to avoid potential misconfiguration).

Reply:
Makes sense. I'll take that branch out.